mono/metadata/sgen-tarjan-bridge.c

```diff
@@ -94,13 +92,21 @@ enum {
 /*
 Optimizations:
 	We can split this data structure in two, those with bridges and those without
+	(and only bridgeless need to record incoming_colors)
 */
+#define API_INDEX_BITS        26
+#define INCOMING_COLORS_BITS  5
+#define API_INDEX_MAX         ((1<<API_INDEX_BITS)-1)
```

Comment:
In this case, the preprocessor can do it:

```
#if defined (API_INDEX_BITS) && defined (INCOMING_COLORS_BITS)
#   if !(API_INDEX_BITS + INCOMING_COLORS_BITS <= 31)
#       error "Assertion failed!"
#   endif
#else
#    error "Someone renamed these #defines and didn't update the assertion."
#endif
```

I don’t suppose we can use C11 `static_assert`.